packages/common-ui-web/src/components/VehiclePartSelection.tsx

```diff
@@ -0,0 +1,93 @@
+import { PartSelectionOrientation, VehiclePart, VehicleType } from '@monkvision/types';
+import { useEffect, useState } from 'react';
+import { useMonkTheme } from '@monkvision/common';
+import { Icon } from '../icons';
+import { VehicleDynamicWireframe, VehicleDynamicWireframeProps } from './VehicleDynamicWireframe';
+
+export interface VehiclePartSelectionProps {
+  vehicleType: VehicleType;
+  orientation?: PartSelectionOrientation;
+  /**
+   * emit when change in the selected parts.
+   */
+  onPartsSelected?: (parts: VehiclePart[]) => void;
+}
+
+/**
+ * Move through different orientations of the vehicle wireframe.
+ * And choose the parts
+ */
+export function VehiclePartSelection({
+  vehicleType,
+  orientation: initialOrientation,
+  onPartsSelected = () => {},
+}: VehiclePartSelectionProps) {
+  const partSelectionOrientations = [
+    PartSelectionOrientation.FRONT_LEFT,
+    PartSelectionOrientation.REAR_LEFT,
+    PartSelectionOrientation.REAR_RIGHT,
+    PartSelectionOrientation.FRONT_RIGHT,
+  ];
+  const [orientation, setOrientation] = useState(
+    initialOrientation ?? partSelectionOrientations[0],
+  );
+  const moveOrientation = (direction: 'next' | 'previous') => {
+    const currentIndex = partSelectionOrientations.indexOf(orientation);
+    const nextIndex =
+      direction === 'next'
+        ? (currentIndex + 1) % partSelectionOrientations.length
+        : (currentIndex - 1 + partSelectionOrientations.length) % partSelectionOrientations.length;
+    setOrientation(partSelectionOrientations[nextIndex]);
+  };
```

Comment:
Instead of this, I think it's easier to read if we have 2 callback functions :

```typescript
const rotateRight = () => {
   ...
};

const rotateLeft = () => {
   ...
};
```